src/daemon/gnet/pool.go

```diff
@@ -435,34 +444,40 @@ func (pool *ConnectionPool) readLoop(conn *Connection, msgChan chan []byte, qc c
 }
 
 func (pool *ConnectionPool) sendLoop(conn *Connection, timeout time.Duration, qc chan struct{}) error {
+	var elapser = elapse.NewElapser(sendLoopDurationThreshold, logger)
 	for {
 		select {
 		case <-pool.quit:
 			return nil
 		case <-qc:
 			return nil
 		case m := <-conn.WriteQueue:
+			elapser.Register("conn.WriteQueue")
```

Comment:
Why not put `defer elapser.CheckForDone()` before the `for` loop like in `readLoop`?

If it needs to be in this block, wrap the block into a function and use the `defer`, instead of calling it for every return